// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

//option java_multiple_files = true;
//option java_package = "io.grpc.examples.helloworld";
//option java_outer_classname = "HelloWorldProto";
//option objc_class_prefix = "HLW";

package mp3;

// The greeting service definition.
service Greeter {

    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply) {}

    // Sends another greeting
    rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}

    rpc begin (beginMessage) returns (beginReply) {}

    //TODO: getValue
    rpc getValue (getMessage) returns (getReply) {}

    rpc setValue (setMessage) returns (setReply) {}

    rpc commit (commitMessage) returns (commitReply) {}

    rpc abort (abortMessage) returns (abortReply) {}
}


// The request message containing the user's name.
message HelloRequest {
    string name = 1;
}
// The response message containing the greetings
message HelloReply {    string message = 1; }

message beginMessage {
    string name = 1;
    string begin = 2;
}
message beginReply {    string message = 1;   }

message getMessage {
    string name = 1;
    string key = 2;
}
message getReply {    string value = 1; }

message setMessage {
    string name = 1;
    string key = 2;
    string value = 3;
}
message setReply {    string key = 1;   }

message commitMessage {
    string name = 1;
    string message = 2;
}
message commitReply {    string reply = 1;  }

message abortMessage {
    string name = 1;
    string message = 2;
}
message abortReply {    string reply = 1;  }
